content/docs/context.md

```diff
@@ -4,220 +4,110 @@ title: Context
+Creates a `{ Provider, Consumer }` pair.
 
-If `contextTypes` is not defined, then `context` will be an empty object.
+Optionally accepts a default value to be passed to Consumers without a Provider ancestor.
 
-## Parent-Child Coupling
+### `Provider`
 
-Context can also let you build an API where parents and children communicate. For example, one library that works this way is [React Router V4](https://reacttraining.com/react-router):
-
-```javascript
-import { BrowserRouter as Router, Route, Link } from 'react-router-dom';
-
-const BasicExample = () => (
-  <Router>
-    <div>
-      <ul>
-        <li><Link to="/">Home</Link></li>
-        <li><Link to="/about">About</Link></li>
-        <li><Link to="/topics">Topics</Link></li>
-      </ul>
-
-      <hr />
-
-      <Route exact path="/" component={Home} />
-      <Route path="/about" component={About} />
-      <Route path="/topics" component={Topics} />
-    </div>
-  </Router>
-);
+```js
+<Provider value={/* some value */}>
 ```
 
-By passing down some information from the `Router` component, each `Link` and `Route` can communicate back to the containing `Router`.
-
-Before you build components with an API similar to this, consider if there are cleaner alternatives. For example, you can pass entire React components as props if you'd like to.
+A React component that allows Consumers to subscribe to context changes.
 
-## Referencing Context in Lifecycle Methods
+Accepts a `value` prop to be passed to Consumers that are descendants of this Provider. One Provider can be connected to many Consumers. Providers can be nested to override values deeper within the tree.
 
-If `contextTypes` is defined within a component, the following [lifecycle methods](/docs/react-component.html#the-component-lifecycle) will receive an additional parameter, the `context` object:
+### `Consumer`
 
-- [`constructor(props, context)`](/docs/react-component.html#constructor)
-- [`componentWillReceiveProps(nextProps, nextContext)`](/docs/react-component.html#componentwillreceiveprops)
-- [`shouldComponentUpdate(nextProps, nextState, nextContext)`](/docs/react-component.html#shouldcomponentupdate)
-- [`componentWillUpdate(nextProps, nextState, nextContext)`](/docs/react-component.html#componentwillupdate)
+```js
+<Consumer>
+  {value => /* render something based on the context value */}
+</Consumer>
+```
 
-> Note:
->
-> As of React 16, `componentDidUpdate` no longer receives `prevContext`.
+A React component that subscribes to context changes.
 
-## Referencing Context in Stateless Functional Components
+Requires a [function as a child](/docs/render-props.html#using-props-other-than-render). The function receives the current context value and returns a React node. All consumers are re-rendered whenever the Provider value changes. Changes are determined by comparing the new and old values using [`Object.is`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is).
```

Comment:
Maybe "using the same algorithm as `Object.is`" so that people don't think it has to be polyfilled.

And let's link directly to Description (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is#Description) because the API of Object.is isn't relevant.